styles/signals.mapcss

```diff
@@ -389,6 +417,74 @@ node|z17-[railway="buffer_stop"]["railway:signal:direction"]["railway:signal:min
 	allow-overlap: true;
 }
 
+/********************************/
+/* DE tram minor stop sign Sh 1 */
+/********************************/
+node|z17-[railway=signal]["railway:signal:direction"]["railway:signal:minor"="DE-BOStrab:sh1"]["railway:signal:minor:form"=sign]
+{
+	z-index: 2000;
+	text: "ref";
+	text-offset: 11;
+	text-color: black;
+	text-halo-radius: 1;
+	text-halo-color: white;
+	font-weight: bold;
+	icon-image: "icons/de/bostrab/sh1-24.png";
+	icon-width: 12;
+	icon-height: 12;
+	text-allow-overlap: true;
+	allow-overlap: true;
+}
+
+/****************************************/
+/* DE tram passing prohibited sign So 5 */
+/****************************************/
+node|z17-[railway=signal]["railway:signal:direction"]["railway:signal:passing"="DE-BOStrab:so5"]["railway:signal:passing:form"=sign]["railway:signal:passing:type"="no_passing"]
+{
+	z-index: 2001;
+	text: "railway:signal:passing:caption";
+	text-color: black;
+	text-halo-radius: 3;
+	text-halo-color: white;
+	text-offset-x: eval(length(tag("railway:signal:passing:caption")) * (0 - 3) - 6);
+	text-offset-y: -12;
+	font-weight: bold;
+	icon-image: "icons/de/bostrab/so5-24.png";
+	icon-width: 12;
+	icon-height: 12;
+	text-allow-overlap: true;
+	allow-overlap: true;
+}
+
+/********************************************/
+/* DE tram passing prohibited end sign So 6 */
+/********************************************/
+node|z17-[railway=signal]["railway:signal:direction"]["railway:signal:passing"="DE-BOStrab:so6"]["railway:signal:passing:form"=sign]["railway:signal:passing:type"="passing_allowed"]
+{
+	z-index: 2001;
+	text: "railway:signal:passing:caption";
+	text-color: black;
+	text-halo-radius: 3;
+	text-halo-color: white;
+	text-offset-x: eval(length(tag("railway:signal:passing:caption")) * (0 - 3) - 6);
+	text-offset-y: -12;
+	font-weight: bold;
+	icon-image: "icons/de/bostrab/so6-24.png";
+	icon-width: 12;
+	icon-height: 12;
+	text-allow-overlap: true;
+	allow-overlap: true;
+}
+
+@media not (user-agent: josm) {
```

Comment:
Please invert this: use extra code for JOSM. Otherwise both code segments will end up in the JavaScript style with one block doing a calculation that is never used.